### PR TITLE
networkit 7.0 (new formula)

### PR DESCRIPTION
**Pull request by fabratu**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Follow-up to #56211 and #51136. Thanks btw for brew in general!

### CODE REVIEW

File: Formula/networkit.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class Networkit < Formula
+  include Language::Python::Virtualenv
+
+  desc "Open source toolkit for large-scale network analysis"
```
Comment (SMillerDev) [inline Formula/networkit.rb:4]: Open source is a requirement for homebrew/core so no need to mention it

File: Formula/networkit.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  homepage "https://networkit.github.io"
+  url "https://github.com/networkit/networkit.git", :tag      => "7.0",
+                                                    :revision => "d8e952f1e4d5e2758e4744e7c6ea7429a59c7cdf"
```
Comment (SMillerDev) [inline Formula/networkit.rb:7]: Please put the tag on a newline

File: Formula/networkit.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  depends_on "cmake"
```
Comment (SMillerDev) [inline Formula/networkit.rb:9]: Is this needed at runtime?

Comment (fabratu) [reply Formula/networkit.rb]: Good point, same is true for cython.

File: Formula/networkit.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
+class Networkit < Formula
+  include Language::Python::Virtualenv
+
+  desc "Performance toolkit for large-scale network analysis"
+  homepage "https://networkit.github.io"
+  url "https://github.com/networkit/networkit.git",
+      :tag => "7.0", :revision => "d8e952f1e4d5e2758e4744e7c6ea7429a59c7cdf"
```
Comment (SMillerDev) [inline Formula/networkit.rb:7]:
```suggestion
  url "https://github.com/networkit/networkit.git",
      :tag => "7.0",
      :revision => "d8e952f1e4d5e2758e4744e7c6ea7429a59c7cdf"
```

File: Formula/networkit.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+           "--rpath=@loader_path;#{rpath_addons}"
+    system Formula["python@3.8"].opt_bin/"python3", "setup.py", "install", "--single-version-externally-managed",
+           "--record=installed.txt", "--prefix=#{libexec}"
+    site_packages = "lib/python#{xy}/site-packages"
+    pth_contents = "import site; site.addsitedir('#{libexec/site_packages}')\n"
+    (prefix/site_packages/"homebrew-networkit.pth").write pth_contents
+  end
+
+  test do
+    system Formula["python@3.8"].opt_bin/"python3", "-c", "import networkit"
```
Comment (SMillerDev) [inline Formula/networkit.rb:31]: Can we test the function as well?

Comment (fabratu) [reply Formula/networkit.rb]: Added a test case for basic functionality

File: Formula/networkit.rb [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+  def install
+    xy = Language::Python.major_minor_version Formula["python@3.8"].opt_bin/"python3"
+    rpath_addons = Formula["libnetworkit"].opt_lib
+
+    ENV.prepend_create_path "PYTHONPATH", libexec+"lib/python#{xy}/site-packages/"
+    system Formula["python@3.8"].opt_bin/"python3", "setup.py", "build_ext", "--networkit-external-core",
+           "--rpath=@loader_path;#{rpath_addons}"
+    system Formula["python@3.8"].opt_bin/"python3", "setup.py", "install", "--single-version-externally-managed",
+           "--record=installed.txt", "--prefix=#{libexec}"
```
Comment (SMillerDev) [inline Formula/networkit.rb:25]: This looks a little cleaner
```suggestion
    system Formula["python@3.8"].opt_bin/"python3", "setup.py", "build_ext",
          "--networkit-external-core",
           "--rpath=@loader_path;#{rpath_addons}"
    system Formula["python@3.8"].opt_bin/"python3", "setup.py", "install",
           "--single-version-externally-managed",
           "--record=installed.txt",
           "--prefix=#{libexec}"
```

**Comment by issyl0** [comment]
I squashed these commits in preparation for merging this. Thanks for your contribution, @fabratu!

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**